analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -52,15 +48,18 @@ private static void DetectTypeChanges(Dictionary<string, RuleDescriptor> rules,
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
 
-            var newRules = items.Where(x => x.ExpectedType is null);
-            newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
+            var newRules = rulesWithUnmatchingType.Where(x => x.ExpectedType is null);
+            newRules
+                .Should()
+                .BeEmpty($": there are rules that exist in '{language}::SonarAnalyzer.RuleCatalog.Rules' file but not in {expectedTypesName}. You need to add them to the latter");
```

Comment:
The error message goes `because xxxx`. So the `: ` should not be needed
```suggestion
                .BeEmpty($"there are rules that exist in '{language}::SonarAnalyzer.RuleCatalog.Rules' file but not in {expectedTypesName}. You need to add them to the latter");
```